src/components/Header/NavComponents/userprofile.tsx

```diff
@@ -30,6 +41,73 @@ export const UserProfile: React.FC<{}> = () => {
   const { data: agaveBalance } = useAmountAvailableToStake(address);
   const userBal = agaveBalance ? bigNumberToString(agaveBalance, 3) : "0";
 
+  const chainAddresses = useChainAddresses();
+
+  // internalAddressesPerNetworkId cannot be indexed for each chain, strigified and parsed back to index
+  var chains =  JSON.parse(JSON.stringify(internalAddressesPerNetworkId))
```

Comment:
Expensive no-op - you can index for each chain by casting it to an indexable, which is implicitly what you're doing here in the most expensive way possible.

Reply:
I didn't understood why `internalAddressPerNetwork` didn't let me get a value directly from a number and it was because I wasn't using the exact type it need, in the end there was no need to really use this